### ISSUE TITLE
Doesn’t provide syntax highlighting except for template strings

### ISSUE THREAD

**Issue by HughxDev** [vokeio/language-javascript-plus#7]
Using Atom 1.30.0 x64 on macOS Sierra 10.12.6:

![](https://i.imgur.com/6aDNUie.png)

**Comment by xeaone** [comment]
Hello @hguiney I do not fully understand. Are you saying that Strings using the `'` apostrophe symbol are not highlighting? Or is the issue that the code surrounding the Template String is not using JavaScript highlighting? 

Can you provide me a full screen shot of Atom with this same file open? (I am mostly wondering what file syntax is being used.)

Also could you copy and paste the code in the screen shot so I can see if the issue persists on my end? 

I am on Arch Linux also using Atom 1.30 x64 and I am not getting any issue with surrounding code not syntax highlighting. If I use a regular String using the apostrophe `'` then I do not get the html/css highlighting but that was by design.

**Comment by HughxDev** [comment]
The issue is that the code surrounding the template string is not using JavaScript highlighting. I’m not using apostrophes.

Here’s a full screenshot. I’m using the JavaScriptPlus syntax. Is it not meant to be used like this?

![](https://i.imgur.com/wKYHPs2.png)

Code snippet:
```js
class VlogmasterUploadSelect extends HTMLElement {
  static get is() {
    return 'vlogmaster-upload-select';
  }

  static get template() {
    return html`
      <template id="${VlogmasterUploadSelect.is}">
        <link rel="stylesheet" href="../components/vlogmaster-upload-select.css" />
        <input id="checkbox" name="" type="checkbox" value="1" checked="checked" />
        <label for="checkbox" class="upload__label" title="Upload">
          <span></span>
          <vlogmaster-provider-name></vlogmaster-provider-name>
        </label>
        <vlogmaster-upload-progress show-percentage simulate-upload></vlogmaster-upload-progress>
        <vlogmaster-identity></vlogmaster-identity>
      </template>
    `;
  }
}
```

**Comment by xeaone** [comment]
Yes I see the issue now. JavaScriptPlus actually should not be selected as a syntax. You can simple use the default JavaScript syntax or Auto Detect because JavaScriptPlus will automatically add its self to the default. So just change your Atom syntax back to Auto Detect or JavaScript and the JavaScriptPlus syntax should work along side it correctly.  Without needing to be manually selected.

I think it would be nice to make JavaScriptPlus syntax include the standard JavaScript highlighting also. But at the time I could not figure out how to do that. 

Let me know if that works or does not work.

**Comment by xeaone** [comment]
@hguiney did this resolve the issue you were experiencing?

**Comment by hguiney** [comment]
Yes it does work now, thank you.

**Comment by HughxDev** [comment]
@AlexanderElias Thanks for advising on proper usage. The issue is partially resolved. I have the Babel ES6 JavaScript language installed, so if I do Auto Detect then Atom defaults to that one, and the string highlighting doesn’t work. But if I manually override to plain JavaScript, I do see the highlighting.

**Comment by xeaone** [comment]
Good to hear. I think I might be able to make it work for that. What is the package link?

**Comment by HughxDev** [comment]
https://atom.io/packages/language-babel
